SIPS/sip-29.md

```diff
@@ -243,13 +269,144 @@ type FungibleAssetMetadata = {
   // List of asset units.
   units: FungibleAssetUnit[];
 };
+```
+### Asset Market Data
+
+```typescript
+type FungibleAssetMarketData = {
+  // Represents a fungible asset market data.
+  fungible: true;
+
+  // The market cap of the asset represented as a decimal number in a string.
+  marketCap?: string;
+
+  // The total volume of the asset represented as a decimal number in a string.
+  totalVolume?: string;
+
+  // The circulating supply of the asset represented as a decimal number in a string.
+  circulatingSupply?: string;
+
+  // The all time high of the asset represented as a decimal number in a string.
+  allTimeHigh?: string;
+
+  // The all time low of the asset represented as a decimal number in a string.
+  allTimeLow?: string;
+
+  pricePercentChange?: {
+    // The `all` value is a special interval that represents all available data.  
+    all?: number;  
+
+    // The interval key MUST follow the ISO 8601 duration format.  
+    [interval: string]: number;
+  };
+};
+```
+
+## Appendix II: Non-fungible Asset Metadata & Market Data
+
+The following asset metadata fields for a non-fungible asset are defined.
+
+### Asset Metadata
+
+`Caip10Address` - A string that represents an address using the [CAIP-10][caip-10] standard.
+
+```typescript
+type NonFungibleAssetCollection = {
+  // Human-friendly name of the asset collection.
+  name: string;
+
+  // The collection address.
+  address: Caip10Address;
+
+  // Ticker symbol of the asset collection.
+  symbol: string;
+
+  // The number of tokens in the collection.
+  tokenCount?: number;
+
+  // The creator address of the asset.
+  creator?: Caip10Address;
+
+  // Base64 data URI or URL representation of the asset icon.
+  imageUrl?: string;
+};
+
+type NonFungibleAssetMetadata = {
+  // Human-friendly name of the asset.
+  name?: string;
+
+  // Ticker symbol of the asset.
+  symbol?: string;
+
+  // Base64 data URI or URL representation of the asset image.
+  imageUrl?: string;
+
+  // The description of the asset.
+  description?: string;
+
+  // Represents a non-fungible asset
+  fungible: false;
+
+  // The time at which the asset was acquired.
+  // The time is represented as a UNIX timestamp. 
+  acquiredAt?: number;
+
+  // Indicates whether the asset is a spam asset.
+  isSpam?: boolean;
```

Comment:
Why would a Snap ever return a spam asset?

Reply:
https://github.com/MetaMask/SIPs/pull/174#discussion_r2095959816

Reply:
```suggestion
  isPossibleSpam?: boolean;
```
Maybe something like this then so that the Snap can flag it as a possible spam token and the client can deal with it based on that?